src/app/content/utils/canonicalUrl.ts

```diff
@@ -9,22 +9,31 @@ export async function getCanonicalUrlParams(
   archiveLoader: AppServices['archiveLoader'],
   osWebLoader: AppServices['osWebLoader'],
   bookId: string,
-  pageShortId: string
+  pageShortId: string,
+  bookVersion?: string
 ) {
   const getBook = makeUnifiedBookLoader(archiveLoader, osWebLoader);
-  const canonicals = [
-    ...(CANONICAL_MAP[bookId] || []),
-    bookId, // use the current book as a last resort
-  ].filter((id) => !!BOOKS[id]);
 
-  for (const id of canonicals) {
-    const version = BOOKS[id].defaultVersion;
+  const canonicals = [...(CANONICAL_MAP[bookId] || []), bookId] // use the current book as a last resort
+    .filter((id) => !!BOOKS[id])
+    .map((id) => ({
+      id,
+      version: bookVersion && id === bookId
+        ? bookVersion
+        : BOOKS[id].defaultVersion,
+    }));
```

Comment:
you could rewrite this as 
```
const canonicals = [
  ...(CANONICAL_MAP[bookId] || []).map(id => ({id, version: BOOKS[id].defaultVersion})),
  {id: bookId, version: bookVersion}
];
```

and it wouldn't need that weird conditional in the map

Reply:
Wouldn't this lead to book-loading functions being possibly called with a book id that is not in `BOOKS`? Filter has to be called before map, but the current book also should be checked (i think there is no reason to possibly get a rejected promise?)

Reply:
lets simplify this whole situation and do 

```
const canonicals = [
    ...(CANONICAL_MAP[bookId] || []),
    ...(bookVersion === BOOKS[bookId].defaultVersion ? [bookId] : []), // use the current book as a last resort if it has the same version as in books config
  ].filter((id) => !!BOOKS[id]);
```

that way we don't have to worry about any non-production urls and its almost the same logic as before

Reply:
Wasn't the point of all this to allow to create a canonical with a version that is a different from the default one if its specified? This makes it skip the current book an because of that it won't be possible to create versioned canonicals, right?

Reply:
in the task i did write `it should use the loaded version` but really i just want to not have an error. since canonicals are always rendered with `openstax.org` urls, and versioned urls don't work on production, i'm realizing it doesn't make _the most_ sense to render that